### PR TITLE
Split session timeout/termination and add new requirement to resolve #1800

### LINKED ISSUE #1800: Opportunistic Session Termination
In #1417 and #1641, we added a new requirement [3.3.6](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x12-V3-Session-management.md#v33-session-termination) related to session termination:

| # | Description | L1 | L2 | L3 | CWE | [NIST &sect;](https://pages.nist.gov/800-63-3/sp800-63b.html) |
| :---: | :--- | :---: | :---: | :---: | :---: | :---: |
| **3.3.6** | [ADDED] Verify that all active sessions are revoked when a user account is disabled or deleted (such as an employee leaving the company). | ✓ | ✓ | ✓ | 613 | |

This joins some existing requirements which also deal with session termination but appear to be focused on either automated functionality after credentials are changed or functionality available to the application user.

| # | Description | L1 | L2 | L3 | CWE | [NIST &sect;](https://pages.nist.gov/800-63-3/sp800-63b.html) |
| :---: | :--- | :---: | :---: | :---: | :---: | :---: |
| **3.3.3** | [MODIFIED, LEVEL L2 > L1] Verify that the application gives the option to terminate all other active sessions after a successful change or removal of any authentication factor (including password change via reset or recovery and, if present, an MFA settings update). This must be effective across the application, federated login (if present), and any relying parties. | ✓ | ✓ | ✓ | 613 | |
| **3.3.4** | Verify that users are able to view and (having re-entered login credentials) log out of any or all currently active sessions and devices. | | ✓ | ✓ | 613 | 7.1 |

Should we create a new requirement to make it clear that an application administrator should also be able to terminate all or some of a user's sessions, if necessary?

### PR DESCRIPTION
**Pull request by tghosth**
This Pull Request relates to issue #1800

### CODE REVIEW

**Comment by tghosth** [comment]
@elarlang note that I deliberately split the commits to make it easier to review

**Comment by elarlang** [comment]
I'll recheck this with fresh brain and validate all "tracking tags". 

First thing to notice is that now there are 2 sections called "Session termination". I think V3.3 should be "Session lifetime" or something like that.

**Review by elarlang — CHANGES_REQUESTED** [review]
https://github.com/OWASP/ASVS/issues/1800#issuecomment-1871442854

File: 5.0/en/0x12-V3-Session-management.md [hunk 1]
```diff
@@ -93,13 +93,17 @@ Attackers start a half-open attack by attempting to lock, reset, or recover a cr
...
-| **3.8.3** | [MOVED FROM 3.3.4] Verify that users are able to view and (having re-entered login credentials) log out of any or all currently active sessions and devices. | | ✓ | ✓ | 613 | 7.1 |
+| **3.8.1** | [MODIFIED, MOVED FROM 3.3.1] Verify that logout and expiration terminate the user's session, such that the back button or a downstream relying party cannot resume an authenticated session. ([C6](https://owasp.org/www-project-proactive-controls/#div-numbering)) | ✓ | ✓ | ✓ | 613 | 7.1 |
+| **3.8.2** | [MODIFIED, LEVEL L2 > L1, MOVED FROM 3.3.3] Verify that the application gives the option to terminate all other active sessions after a successful change or removal of any authentication factor (including password change via reset or recovery and, if present, an MFA settings update). | ✓ | ✓ | ✓ | 613 | |
+| **3.8.3** | [MOVED FROM 3.3.4] Verify that users are able to view and (having re-entered login credentials) terminate any or all currently active sessions and devices. | | ✓ | ✓ | 613 | 7.1 |
```
Comment (elarlang) [inline 5.0/en/0x12-V3-Session-management.md:104]: add MODIFIED tag
this is the only requirement to mention devices - doing so, it is hidden requirement for tracking devices. I prefer to remove "and devices".

Comment (jmanico) [reply 5.0/en/0x12-V3-Session-management.md:104]: In a traditional session we "terminate" that session. In a JWT stateless mechanism, we "revoke" a session. Can we please change "terminate" to "terminate or revoke" a session to account for both architecture types?

Comment (elarlang) [reply 5.0/en/0x12-V3-Session-management.md:104]: my point of view: "terminate session" is logic and functionality to provide. "revoke" is a technical term to make JWT invalid, and from a functionality point of view, it means terminating a session, although it is not called a session (it is still a server side state handling).

Comment (tghosth) [reply 5.0/en/0x12-V3-Session-management.md:104]: Addressed these three comments including adding a comment at the start of the section.

File: 5.0/en/0x12-V3-Session-management.md [hunk 2]
```diff
@@ -30,21 +30,19 @@ As previously noted, these requirements have been adapted to be a compliant subs
 
 TLS or another secure transport channel is mandatory for session management. This is covered in the Communications Security chapter.
 
-## V3.3 Session Termination
+## V3.3 Session Timeouts
```
Comment (elarlang) [inline 5.0/en/0x12-V3-Session-management.md:33]: To align with the style of the rest of the titles, I think it should be "Session Timeout"

**Pull request merged**